src/core/client/admin/routes/Moderate/SiteSelector/SiteSelectorContainer.tsx

```diff
@@ -22,7 +24,37 @@ interface Props {
 }
 
 const SiteSelectorContainer: React.FunctionComponent<Props> = (props) => {
+  const context = useCoralContext();
   const [loadMore, isLoadingMore] = useLoadMore(props.relay, 10);
+  // Buffering in the sense that we gather changes and then "flush"
+  // them 1 second after the last change is received
+  const [filterBuffer, setFilterBuffer] = useState<string | null>(null);
+  const [searchFilter, setSearchFilter] = useState<string | null>(null);
+  const [searchTimeout, setSearchTimeout] = useState<number | undefined>();
+
+  const [, isRefetching] = useRefetch(props.relay, 10, {
+    searchFilter: searchFilter || null,
+  });
+
+  useEffect(() => {
+    if (filterBuffer === searchFilter) {
+      return;
+    }
+
+    if (searchTimeout) {
+      context.window.clearTimeout(searchTimeout);
+    }
+
+    const newTimeout = context.window.setTimeout(
+      () => setSearchFilter(filterBuffer),
+      1000
+    );
+
+    setSearchTimeout(newTimeout);
+
+    return () => context.window.clearTimeout(searchTimeout);
+  }, [filterBuffer, context.window]);
+  // marcushaddon: omitting searchFilter from dependency array to avoid an infinite update loop
```

Comment:
Thanks for doing this. I noticed something else going on if the query to filter/paginate errors. When there is an error, the page spams out thousands of requests as fast as possible, hammering the API.

A good way to test this is without an index for Site (seems to be missing in PR description notes) try and filter. Mongo will error cause it can't do text searches without a text index. Open the dev tools console and you'll see the page is absolutely hammering the API with retries to filter the site list.

Reply:
Do we have the ability to specify a rate limit on the front end using PaginationContainers? I'm not seeing any examples of doing this on a per-request basis.

Reply:
Yeah, if you look into the `AutoLoadMore` there is a `useEffect` that is performing the `onLoadMore`. Could stick a retry counter or use `setTimeOut`'s to back off the reloads I think? I could be wrong though, if so, then something higher up is re-calling the load events and you can back it off there instead.